third_party/terraform/website/docs/r/bigquery_dataset_iam.html.markdown

```diff
@@ -0,0 +1,120 @@
+---
+layout: "google"
+subcategory: "BigQuery"
+page_title: "Google: google_bigquery_dataset_iam"
+sidebar_current: "docs-google-bigquery-dataset-iam"
+description: |-
+ Collection of resources to manage IAM policy for a BigQuery dataset.
+---
+
+# IAM policy for BigQuery dataset
+
+Three different resources help you manage your IAM policy for BigQuery dataset. Each of these resources serves a different use case:
+
+* `google_bigquery_dataset_iam_policy`: Authoritative. Sets the IAM policy for the dataset and replaces any existing policy already attached.
+* `google_bigquery_dataset_iam_binding`: Authoritative for a given role. Updates the IAM policy to grant a role to a list of members. Other roles within the IAM policy for the dataset are preserved.
+* `google_bigquery_dataset_iam_member`: Non-authoritative. Updates the IAM policy to grant a role to a new member. Other members for the role for the dataset are preserved.
+
+~> **Note:** These resources **cannot** be used with `google_bigquery_dataset_access` resources or the `access` field on `google_bigquery_dataset` or they will fight over what the policy should be.
+
+~> **Note:** Using any of these resources will remove any authorized view permissions from the dataset. To assign and preserve authorized view permissions use the `google_bigquery_dataset_access` instead.
+
+~> **Note:** Legacy BigQuery roles `OWNER` `WRITER` and `READER` **cannot** be used with any of these IAM resources. Instead use the full role form of: `roles/bigquery.dataOwner` `roles/bigquery.dataEditor` and `roles/bigquery.dataViewer`.
+
+~> **Note:** `google_bigquery_dataset_iam_policy` **cannot** be used in conjunction with `google_bigquery_dataset_iam_binding` and `google_bigquery_dataset_iam_member` or they will fight over what your policy should be.
+
+~> **Note:** `google_bigquery_dataset_iam_binding` resources **can be** used in conjunction with `google_bigquery_dataset_iam_member` resources **only if** they do not grant privilege to the same role.
+
+## google\bigquery\_dataset\_iam\_policy
+
+```hcl
+data "google_iam_policy" "admin" {
+  binding {
+    role = "roles/editor"
+
+    members = [
+      "user:jane@example.com",
+    ]
+  }
+}
+
+resource "google_bigquery_dataset_iam_policy" "dataset" {
+  dataset_id  = "your-dataset-id"
+  policy_data = data.google_iam_policy.admin.policy_data
+}
+```
+
+## google\_bigquery\_dataset\_iam\_binding
+
+```hcl
+resource "google_bigquery_dataset_iam_binding" "dataset" {
+  dataset_id = "your-dataset-id"
+  role       = "roles/editor"
```

Comment:
not necessary but maybe change one of these to the bigquery roles (dataOwner or dataEditor)